core/domain/user_domain.py

```diff
@@ -448,4 +445,18 @@ def from_raw_dict(cls, raw_user_data_dict):
             ModifiableUserData. The domain object representing the user data
             dict transformed according to the latest schema version.
         """
-        return ModifiableUserData.from_dict(raw_user_data_dict)
+        data_schema_version = raw_user_data_dict.get('schema_version')
+
+        if data_schema_version is None:
+            raise Exception(
+                'Invalid modifiable user data: no schema version specified.')
+        if (data_schema_version < 1) or (
+                data_schema_version > cls.CURRENT_SCHEMA_VERSION):
+            raise Exception(
+                'Invalid version %s received. At present we can only process v1'
+                ' to v%s modifiable user data.' % (
+                    data_schema_version, cls.CURRENT_SCHEMA_VERSION)
+            )
+
+        return (
+            cls.from_dict(raw_user_data_dict), data_schema_version)
```

Comment:
Don't you need to change the Returns docstring if you do this?

Also I'm a bit confused. Why do we need to return the data_schema_version? I think that this function should just forward-migrate the domain object to the latest schema version, and it should be understood that any object this returns will have the latest schema version (which is basically what the current docstring says). This will simplify things because any other method/function using the object can assume that it's the "latest" one.

Reply:
I actually agree with you, I do want the function to return the object of latest schema always (and not any other schema version which has to be separately mentioned). In the code, I returned schema version for a double check thing to verify that the finally the version is indeed the latest one, and it was helping me test this behaviour (although if we see it's like a redundant check only, I am already checking other attributes' values and that the new field is being correctly added).

I will remove it because what you said makes sense!
> This will simplify things because any other method/function using the object can assume that it's the "latest" one.

Reply:
Edit: @seanlip did the changes now.